### PR TITLE
[JENKINS-52009] Added support for timezone in cron and percent sign in parameter key/value

### PR DESCRIPTION
**Pull request by shbhardwaj**
**Current behavior** : No way to set timezone for all the parameterized cron expressions.
**After this change** : Timezone in first line will be applied to all the cron expressions see below for example.
TZ=Australia/Sydney
H/15 * * * * %name=value
H(0-29)/10 * * * * % name=value; othername=othervalue

The timezone Australia/Sydney will be applied for both the crons.

**Current behavior** : If percent sign is part of some parameter key or value then all the parameters of the job are ignored.
**After this change** : Only the first % sign will be used to separate the cron expression from the parameters. Any other percent sign in key/value will be part of the key/value see below for example

H/15 * * * * %name=value;key=10%;
The parameters will be name=value, key=10%

### CODE REVIEW

**Comment by shbhardwaj** [comment]
The hpi file generated for this PR can be found at 
https://ci.jenkins.io/job/Plugins/job/parameterized-scheduler-plugin/job/PR-47/1/artifact/target/parameterized-scheduler.hpi

Use hpi file to upload the plugin with above mentioned features.

File: src/test/java/org/jenkinsci/plugins/parameterizedscheduler/ParameterizedCronTabTest.java
Comment (andymcf) [inline src/test/java/org/jenkinsci/plugins/parameterizedscheduler/ParameterizedCronTabTest.java]: spelling should be **separator** not seperator

great PR otherwise - I came here looking for a way to use timezone :)

**Comment by bawjensen** [comment]
Question for the maintainers: Would it be quicker to get this timezone functionality merged if there were a separate PR with just that functionality? I can create that PR, if so, given that's what my team has the most use for (granted I would be shamelessly copying @shbhardwaj's code to do so)

**Comment by jef** [comment]
> Question for the maintainers: Would it be quicker to get this timezone functionality merged if there were a separate PR with just that functionality? I can create that PR, if so, given that's what my team has the most use for (granted I would be shamelessly copying @shbhardwaj's code to do so)

This should happen or merge this! Long over due.

**Comment by chdwlch** [comment]
Would be great to have this merged in!

**Comment by balaarunreddyv1** [comment]
please merge it.

**Comment by davidmoralc** [comment]
Hi @batmat it would be great to review and merge the changes. Do you think a different pull request only for timezone should be needed as @bawjensen suggested to have the release faster? cc: @shbhardwaj

**Comment by sichao-uber** [comment]
Can anyone merge this?

**Comment by nvwls** [comment]
@batmat Please advise if there is anything that needs to be done to get this merged.

This is really important for orgs that have a centralized jenkins installation and teams spread across the world. Developers want their jobs to run in their local time and this helps deal with Daylight Savings. Not all time zones switch at the same time and scheduled jobs that are expected to run at a given time are now running an hour earlier or later depending on the time of the year.

**Comment by genehynson** [comment]
let's get this merged!

**Comment by artem-kosenko** [comment]
merge this feature please. ppl need it

**Comment by hestrompa** [comment]
Can anyone merge this please?? This would be a great fix!

File: src/test/java/org/jenkinsci/plugins/parameterizedscheduler/ParameterizedCronTabTest.java
Comment (mpasat) [reply src/test/java/org/jenkinsci/plugins/parameterizedscheduler/ParameterizedCronTabTest.java:62]: Hi @shbhardwaj 
Did you have chance to look at above comment?
Thanks!

**Comment by chunming08** [comment]
@shbhardwaj Can you please kindly resolve the conflict and update your PR? Then @res0nance, if you would be able to merge this into the master at an early convenience, it would be great!
I am looking forward to having the change in 0.9.3 before my team's Jenkins instance upgrade. Much appreciated!

File: src/test/java/org/jenkinsci/plugins/parameterizedscheduler/ParameterizedCronTabTest.java
Comment (shbhardwaj) [reply src/test/java/org/jenkinsci/plugins/parameterizedscheduler/ParameterizedCronTabTest.java]: spelling fixed

**Comment by shbhardwaj** [comment]
@andymcf @mpasat Spelling fixed
@chunming08 Updated the PR and resolved the conflicts
I hope this can be merged soon its more than a year that this PR is open.

**Review by res0nance — APPROVED** [review]
@shbhardwaj sorry about that, I recently took over as maintainer roughly 2 weeks ago

**Comment by mikeprigodich** [comment]
What is the ETA for making this fix available in a new release? We would like to make use of it.

**Comment by res0nance** [comment]
> What is the ETA for making this fix available in a new release? We would like to make use of it.

Waiting for a reply on #44 followed by some testing, then a release. Can't give an exact ETA, but maybe a week or two after that PR is merged.

**Comment by waded** [comment]
It would be nice if README had an example usage that included timezone, e.g.

        parameterizedCron('''
            TZ=America/Denver
            */2 * * * * %GREETING=Hola;PLANET=Pluto
            */3 * * * * %PLANET=Mars
        ''')

(FYI, currently the TZ expression can only be on the first line, applies to all lines if present.)

**Pull request merged**
